server/src/test/java/io/druid/server/coordination/coordination/BatchDataSegmentAnnouncerTest.java

```diff
@@ -169,13 +178,30 @@ public void testSingleAnnounce() throws Exception
   }
 
   @Test
+  public void testSingleAnnounceManyTimes() throws Exception
+  {
+    int prevMax = maxBytesPerNode.get();
+    maxBytesPerNode.set(2048);
+    try {
+      for (DataSegment segment : testSegments) {
+        segmentAnnouncer.announceSegment(segment);
+      }
+      List<String> zNodes = cf.getChildren().forPath(testSegmentsPath);
+      Assert.assertEquals(17, zNodes.size());
```

Comment:
Please add a comment of why is the zNodes size expected to be 17. 
Also, might be worth verifying that all 100 are being announced.